src/ui/pages/Menu/components/ConnectWallet/ConnectWallet.tsx

```diff
@@ -116,21 +118,21 @@ const ConnectWallet = forwardRef<ConnectWalletOptionRef, object>(
       handleOpenVerify();
     };
 
-    const handleDeleteConnection = (data: ConnectionData) => {
+    const handleDeleteConnection = async (data: ConnectionData) => {
       actionInfo.current = {
         type: ActionType.None,
       };
-
-      // TODO: Implement delete wallet connection logic
-
+      await Agent.agent.peerConnectionMetadataStorage.deletePeerConnectionMetadataRecord(
+        data.id
+      );
+      const connections =
+        await Agent.agent.peerConnectionMetadataStorage.getAllPeerConnectionMetadata();
+      dispatch(setWalletConnectionsCache(connections));
       dispatch(setToastMsg(ToastMsgType.WALLET_CONNECTION_DELETED));
     };
 
     const handleConnectWallet = () => {
```

Comment:
Hmm we aren't calling connect from here, only `Connect.tsx`. Can you ask Duke why we have 2 places to show the connect toast message?

Reply:
I think he thought the connection could be made after scanning the meerkat.

Reply:
Maybe @Sotatek-DukeVu can push or suggest a small fix to this PR to correctly show the toast once, and maybe rename this function

Reply:
Can you check with Duke on this one

Reply:
After we connect with new connection, we will cache it in `Connection history` and this function is using to connect with cached connection without scan QR or paste id.

Reply:
Well, We need confirm this flow. Because with the information I got from Sal, ticket requirement and design We can connect by click connect/disconnect button.

1. https://cardanofoundation.atlassian.net/browse/DTIS-821
2. https://www.figma.com/design/XN6K76yt13zxCUHPh1Egr8/Identity-wallet?node-id=7770-118942&t=YiuydYk5DYcrOECE-0

Reply:
But you are right, I think we should choose identifier before we connect to wallet.

Reply:
@Sotatek-DukeVu The identifier must be selected before we actually connect to the dApp. When you are disconnected from an old connection and try to re-connect, this shows the identifier selection screen again (with the previous identifier already selected) - it still only re-connects to the dApp once the identifier is confirmed by the user.

Reply:
I double checked with Robert. So yes we should only consider the wallet connected once we have selected the identifier and actually connected via Meerkat.

So my understanding is:
1. When initially connecting, we paste the Meerkat ID, select the identifier and _then_ after connecting with Meerkat we show the toast to say we've connected.
2. If we are disconnected and would like to re-connect - when we click Connect, we will show the identifier list and after selecting the identifier (the identifier from when it was previously connected should be pre-selected in the list) we will connect with Meerkat and show the toast.
3. If we disconnect here, we will show the disconnect toast since there is no identifier selection to be done. I'm not sure if this requires a pin code to be entered @Sotatek-DukeVu ?

@Sotatek-HocNguyena Can you make any necessary changes to make sure that the toasts and connected dApp are updated only after selecting the identifier?

Reply:
`If we disconnect here, we will show the disconnect toast since there is no identifier selection to be done. I'm not sure if this requires a pin code to be entered @Sotatek-DukeVu ?`

I remember we discussed and confirmed that we will remove the pin/pass code input action for the connect, add new connection and We only use the passcode/pin when we delete the connection.
cc: @robsalaman

Reply:
Is the pincode verification already in place for deleting now @Sotatek-DukeVu ?

Reply:
Yes. I implemented it.

Reply:
thanks :)